File: p2/P2/java/InterfoneResidencial.java
public class InterfoneResidencial extends Interfone {
    String nome_proprietario, telefone_proprietario;

    public InterfoneResidencial(String modelo, String endereco, String nome_proprietario, String telefone_proprietario) {
        super(modelo, endereco);
        this.nome_proprietario = nome_proprietario;
        this.telefone_proprietario = telefone_proprietario;
    }

    public InterfoneResidencial cadastro(String modelo, String endereco, String nome_proprietario, String telefone_proprietario) {
        return new InterfoneResidencial(modelo, endereco, nome_proprietario, telefone_proprietario);
    }

    @Override
    public void ligar_para_proprietario() {
        System.out.println("\n Ligando para o \"" + this.nome_proprietario + "\" no numero \"" + this.telefone_proprietario + "\", do endereco \"" + this.endereco + "\" ...");
    }

    @Override
    public void registrar_chamada() {
        System.out.println("\n Chamada registrada as 14:35 ...\n Duracao: 00:02:15 ...");
    }
    
    @Override
    public String toString() {
        return "\n Modelo: " + this.modelo + "\n Endereco: " + this.endereco + "\n Nome do proprietario: " + this.nome_proprietario + "\n Telefone do proprietario: " + this.telefone_proprietario;
    }
}

File: p2/python/InterfoneResidencial.py
from Interfone import Interfone

class InterfoneResidencial(Interfone):
    def __init__(self, modelo, endereco, nome_proprietario, telefone_proprietario):
        super().__init__(modelo, endereco)
        self.nome_proprietario = nome_proprietario
        self.telefone_proprietario = telefone_proprietario

    @classmethod
    def cadastro(cls):
        modelo = input('\n Modelo: ')
        endereco = input(' Endereco: ')
        nome_proprietario = input(' Nome do proprietario: ')
        telefone_proprietario = input(' Telefone do proprietario: ')
        return cls(modelo, endereco, nome_proprietario, telefone_proprietario)

    def ligar_para_proprietario(self):
        print(f'\n Ligando para o "{self.nome_proprietario}" no numero "{self.telefone_proprietario}", do endereco "{self.endereco}"...')

    def registrar_chamada(self):
        print(f'\n Chamada registrada as 14:35 ...\n Duracao: 00:02:15 ...')

    def __str__(self):
        return f'\n Modelo: {self.modelo}\n Endereco: {self.endereco}\n Nome do proprietario: {self.nome_proprietario}\n Telefone do proprietario: {self.telefone_proprietario}'
